crates/ruff/src/settings/mod.rs

```diff
@@ -232,7 +232,10 @@ impl Settings {
             flake8_gettext: config.flake8_gettext.map(Into::into).unwrap_or_default(),
             isort: config.isort.map(Into::into).unwrap_or_default(),
             mccabe: config.mccabe.map(Into::into).unwrap_or_default(),
-            pep8_naming: config.pep8_naming.map(Into::into).unwrap_or_default(),
+            pep8_naming: match config.pep8_naming {
+                Some(config) => config.try_into()?,
+                None => Default::default(),
+            },
```

Comment:
If you rebase, you'll see we have a similar pattern for `copyright` now:

```rust
copyright: config
    .copyright
    .map(copyright::settings::Settings::try_from)
    .transpose()?
    .unwrap_or_default(),
```

Consider re-using that snippet here, for consistency?